pkg/antctl/client.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package antctl
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+// RequestOption describes options to issue requests to the antctl server.
+type RequestOption struct {
+	GroupVersion *schema.GroupVersion
+	// Path is the URI the ongoing request
+	Path string
+	// Kubeconfig is the path to the config file for kubectl.
+	Kubeconfig string
+	// Name is the command which is going to be requested.
+	Name string
+	// Args are the parameters of the ongoing request.
+	Args map[string]string
+	// Timeout specifies a time limit for requests made by the client. The timeout
+	// includes connection time, any redirects, and reading the response body.
+	TimeOut time.Duration
+}
+
+// client issues requests to an antctl server and gets the response.
+type client struct {
+	// inPod tells the running mode of the client.
+	inPod bool
+	// codec is the CodecFactory for this command, it is needed for remote accessing.
+	codec serializer.CodecFactory
+}
+
+// resolveKubeconfig tries to load the kubeconfig specified in the RequestOption.
+// It will return error if the stating of the file failed or the kubeconfig is malformed.
+// It will not try to look up InCluster configuration. If the kubeconfig is loaded,
```

Comment:
Why don't try InCluster config?

Reply:
It was not considered in current implementation, but I'm going to implement an image to run antctl in pod.

Reply:
Use #310 to track this issue.